src/sources/util/message_decoding.rs

```diff
@@ -27,49 +29,32 @@ pub fn decode_message<'a>(
-                                    path!(schema.source_type_key()),
-                                    path!("source_type"),
-                                    Bytes::from(source_type),
-                                );
-                                match log_namespace {
-                                    LogNamespace::Vector => {
-                                        if let Some(timestamp) = timestamp {
-                                            log.try_insert(
-                                                metadata_path!(source_type, "timestamp"),
-                                                timestamp,
-                                            );
-                                        }
+            Ok(Some((events, _))) => Some(events.into_iter().map(move |mut event| {
+                if let Event::Log(ref mut log) = event {
+                    log_namespace.insert_vector_metadata(
+                        log,
+                        path!(schema.source_type_key()),
+                        path!("source_type"),
+                        Bytes::from(source_type),
+                    );
+                    match log_namespace {
+                        LogNamespace::Vector => {
+                            if let Some(timestamp) = timestamp {
+                                log.try_insert(metadata_path!(source_type, "timestamp"), timestamp);
+                            }
 
-                                        log.insert(
-                                            metadata_path!("vector", "ingest_timestamp"),
-                                            now,
-                                        );
-                                    }
-                                    LogNamespace::Legacy => {
-                                        if let Some(timestamp) = timestamp {
-                                            log.try_insert(schema.timestamp_key(), timestamp);
-                                        }
-                                    }
-                                }
+                            log.insert(metadata_path!("vector", "ingest_timestamp"), now);
+                        }
+                        LogNamespace::Legacy => {
+                            if let Some(timestamp) = timestamp {
+                                log.try_insert(schema.timestamp_key(), timestamp);
                             }
-                            event
-                        })
-                        .fold_finally(
-                            0,
-                            |size, event: &Event| size + event.estimated_json_encoded_size_of(),
-                            move |byte_size| emit!(EventsReceived { byte_size, count }),
-                        ),
-                )
-            }
+                        }
+                    }
+                }
+                events_received.emit(CountByteSize(1, event.estimated_json_encoded_size_of()));
```

Comment:
It looks like `count` used to be taken from the `events.len()` before your changes?
Why is it changing to a hard coded ` here? Is it always going to be one event? If so why do we need the `into_iter()` above?

Reply:
Ah, I see it now. Thanks for the clarity. Need to use that ignore whitespace feature more often.